app/controllers/OrderController.java

```diff
@@ -37,24 +46,15 @@ public Result show() {
     }
 
     public Result submit() {
-        try {
-            final Cart currentCart = cartService.getOrCreateCart(session());
-            final Cart clearedCart = cartService.clearCart(currentCart);
-            return ok(success.render());
-        } catch (Exception e) {
-            Logger.error(e.getMessage(), e);
-            return internalServerError();
-        }
+        LOG.debug("Submitting Order details page");
+        final Cart currentCart = cartService.getOrCreateCart(session());
+        cartService.clearCart(currentCart);
```

Comment:
The method was trying to re-use the same cart, but I think it's better if we just drop it and create a new one. But only if we are not creating a new cart in the home page.